packages/truffle-core/lib/test.js

```diff
@@ -105,15 +106,22 @@ const Test = {
       runner
     );
 
-    await this.setJSTestGlobals(web3, accounts, testResolver, runner);
+    await this.setJSTestGlobals({
+      config,
+      web3,
+      accounts,
+      testResolver,
+      runner,
+      compilation: compilations.solc
+    });
 
     // Finally, run mocha.
     process.on("unhandledRejection", reason => {
       throw reason;
     });
 
     return new Promise(resolve => {
-      mocha.run(failures => {
+      this.mochaRunner = mocha.run(failures => {
```

Comment:
`this.mochaRunner` isn't block scoped here?

Reply:
I don't follow. What do you mean?

Reply:
`this.mochaRunner` is instantiated inside `Test.run`, but then called within `setJSTestGlobals.

seems like it would be `undefined` within `setJS...`

Reply:
Oh, I see; yes that is confusing.

It's fine because `this.mochaRunner` is accessed inside the `async operation => { ... }` closure, so lookup happens when `debug()` is called, v. likely giving enough time for `Test.run` to reach this line. Ugly stateful business, isn't it?

Any ideas on how to improve this?

Reply:
how about a comment?